src/utility.jl

```diff
@@ -74,3 +75,260 @@ function visualize_flow(method::ColorBased, flow::Array{SVector{2, Float64}, 2};
+"""
+function evaluate_flow_error(ground_truth_flow::Array{SVector{2, Float64}, 2}, estimated_flow::Array{SVector{2, Float64}, 2}, error_type::AngularError)
+    result = Array{Float64, 2}(undef, size(estimated_flow))
+    for i in 1:size(estimated_flow)[1]
+        for j in 1:size(estimated_flow)[2]
+            if is_flow_known(estimated_flow[i, j]) && is_flow_known(ground_truth_flow[i, j])
+                p1 = append!(convert(Array{Float64, 1}, ground_truth_flow[i, j]), 1.0)
+                p2 = append!(convert(Array{Float64, 1}, estimated_flow[i, j]), 1.0)
+                cosine = dot(p1, p2)/(norm(p1)*norm(p2))
+                if cosine > 1
+                    result[i, j] = 0
+                else
+                    result[i, j] = acos(cosine)
+                end
+                if result[i, j] <= 1e-5
+                    result[i, j] = 0
+                end
+            else
+                result[i, j] = NaN
+            end
+        end
+    end
+    return result
+end
+
+
+"""
+```
+calculate_statistics(error)
+```
+Returns mean and standard deviation, as float values, RX and AX statistics, as dictionaries, for the error input flow.
+
+# Details
+RX and AX are the robustness statistics. RX denotes the percentage of pixels that have
+an error measure over X. AX denotes the accuracy of the error measure at the Xth percentile.
+
+# Arguements
+The error parameters needs to be two-dimensional arrays of length-2 vectors (of type SVector) 
+which represent the error between two flows.
+
+# Example 
+
+Compute the Mean, SD, RX, AX stats of the flow error calculated using endpoint error method.
+```julia
+
+using ImageTracking
+
+error = evaluate_flow_error(ground_truth_flow, estimated_flow, EndpointError())
+mean, sd, rx, ax = calculate_statistics(error) 
+```
+
+Compute the mean, SD, RX, AX stats of the flow error calculated using angula error method.
+```julia
+
+using ImageTracking
+
+error = evaluate_flow_error(ground_truth_flow, estimated_flow, AngularError())
+mean, sd, rx, ax = calculate_statistics(error) 
+```
+
+# References
+[1] S. Baker, D. Scharstein, JP Lewis, S. Roth, M.J. Black, and R. Szeliski. A database and
+evaluation methodology for optical flow.International Journal of Computer Vision, 92(1):1–31, 2011.
+"""
+function calculate_statistics(error::Array{Float64, 2})
+    R_thresholds = [0.5, 1.0, 2.0, 3.0, 5.0]
+    A_thresholds = [0.5, 0.75, 0.95]
+    
+    # mean and standard deviations
+    mean_value = mean(skipmissing(error))
+    std_value = std(skipmissing(error); mean=mean_value)
+
+    # RX stats
+    RX_count = Dict()
+    for rx in R_thresholds
+        count_above_thresh = count(x->(x > rx), error)
+        RX_count[rx] = count_above_thresh / (size(error)[1]*size(error)[2])
+    end
+    
+    # AX stats
+    AX_count = Dict()
+    hist = imhist(error)
+    max_val = maximum(hist[2])
+    total_pix = size(error)[1]*size(error)[2]
+    pixel_cumulative_count = cumsum(hist[2])
```

Comment:
You can get rid of the `hist[2]` and `hist[1]` below because you now have `edges` and `counts`.